packages/language/src/validation/messages/IBM1219I-leave-exits-noniterative-do.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+import { PLICodes } from ".";
+import { Severity, tokenToRange, tokenToUri } from "../../language-server/types";
+import * as AST from "../../syntax-tree/ast";
+import { ValidationAcceptor } from "../validator";
+
+/**
+ * IBM1219I – LEAVE will exit noniterative DO.
+ *
+ * Warn when an unlabeled LEAVE sits inside a non-iterative DO
+ * that itself is nested within an iterative DO-loop. In this case the
+ * LEAVE exits only the group, not the loop — often not what the author intended.
+*/
+export function IBM1219I_leave_exits_noniterative_do(
+    node: AST.LeaveStatement,
+    acceptor: ValidationAcceptor,
+): void {
+    // If LEAVE has a label, rule does not apply.
+    if (node.label) return;
+
+    // Find nearest enclosing DO
+    let ancestor: AST.SyntaxNode | null = node.container;
+    let nearestDo: AST.DoStatement | undefined;
+    while (ancestor) {
+        if (ancestor.kind === AST.SyntaxKind.DoStatement) {
+            nearestDo = ancestor as AST.DoStatement;
+            break;
+        }
+        ancestor = (ancestor as AST.AstNode).container;
```

Comment:
Suggestion: `container` exists as a property on every node. No need to do any casting.
```suggestion
        ancestor = ancestor.container;
```